src/analysis/_testData/testData.ts

```diff
@@ -824,7 +830,7 @@ export const azureRuntime: ListRuntimeItem = {
     machineType: 'Standard_DS2_v2',
     persistentDiskId: 16902,
     region: 'eastus',
-  },
+  } satisfies AzureConfig,
```

Comment:
I suspect the Runtime model types could be improved to enforce that the cloud provider for runtimeConfig matches cloudContext.

Reply:
I did notice a few types that needed improvint/tightening.... did a bit in this PR and hope to do more soon as I get comfortable with the code area.